pkg/controller/iam/serviceaccountkey.go

```diff
@@ -0,0 +1,274 @@
+	errCreateServiceAccountKey = "cannot create GCP ServiceAccountKey object via IAM API"
+	errDeleteServiceAccountKey = "cannot delete GCP ServiceAccountKey object via IAM API"
+	// format string for the relative resource names of ServiceAccountKeys built upon rrns of ServiceAccounts
+	fmtKeyRelativeResourceName = "%s/keys/%s"
+	// format string for invalid service account reference errors
+	fmtErrInvalidServiceAccountRef = "invalid service account reference: %v"
+	// connection detail keys
+	keyPrivateKeyType = "privateKeyType"
+	keyPrivateKeyData = "privateKey"
+	keyPublicKeyType  = "publicKeyType"
+	keyPublicKeyData  = "publicKey"
+)
+
+// SetupServiceAccountKey adds a controller that reconciles ServiceAccountKeys.
+func SetupServiceAccountKey(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter) error {
+	name := managed.ControllerName(v1alpha1.ServiceAccountKeyGroupKind)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewDefaultManagedRateLimiter(rl),
+		}).
+		For(&v1alpha1.ServiceAccountKey{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.ServiceAccountKeyGroupVersionKind),
+			managed.WithInitializers(),
+			managed.WithExternalConnecter(&serviceAccountKeyServiceConnector{client: mgr.GetClient()}),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type serviceAccountKeyServiceConnector struct {
+	client client.Client
+}
+
+// Connect sets up SA key external client using credentials from the provider
+func (c *serviceAccountKeyServiceConnector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	_, opts, err := gcp.GetAuthInfo(ctx, c.client, mg)
+
+	if err != nil {
+		return nil, err
+	}
+
+	s, err := iamv1.NewService(ctx, opts)
+
+	if err != nil {
+		return nil, errors.Wrap(err, errNewClient)
+	}
+
+	return &serviceAccountKeyExternalClient{
+			serviceAccountKeyClient: s.Projects.ServiceAccounts.Keys,
+		},
+		errors.Wrap(err, errNewClient)
+}
+
+type serviceAccountKeyExternalClient struct {
+	serviceAccountKeyClient serviceaccountkey.Client
+}
+
+func (s *serviceAccountKeyExternalClient) Observe(ctx context.Context,
+	mg resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mg.(*v1alpha1.ServiceAccountKey)
+
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotServiceAccountKey)
+	}
+
+	rrn, extNameExists, err := resourcePath(cr)
+
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errGetServiceAccountKey)
+	}
+
+	if !extNameExists {
+		return managed.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+
+	getCall := s.serviceAccountKeyClient.Get(rrn)
+
+	if cr.Spec.ForProvider.PublicKeyType != nil && *cr.Spec.ForProvider.PublicKeyType != "" {
+		getCall = getCall.PublicKeyType(*cr.Spec.ForProvider.PublicKeyType)
+	}
```

Comment:
```suggestion
	getCall := s.serviceAccountKeyClient.Get(rrn).PublicKeyType(gcpclients.StringValue(cr.Spec.ForProvider.PublicKeyType))
```

Would that cause any problem here?